faiss/impl/Panorama.h

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// -*- c++ -*-
+
+#ifndef FAISS_PANORAMA_H
+#define FAISS_PANORAMA_H
+
+#include <faiss/impl/IDSelector.h>
+#include <faiss/impl/PanoramaStats.h>
+#include <faiss/utils/distances.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <vector>
+
+namespace faiss {
+
```

Comment:
Please add a documentation block here (if it's redundant with IndexFlatPanorama redirect to it)